test/parallel/test-dotenv-edge-cases.js

```diff
@@ -10,17 +10,15 @@ const validEnvFilePath = '../fixtures/dotenv/valid.env';
 const nodeOptionsEnvFilePath = '../fixtures/dotenv/node-options.env';
 
 describe('.env supports edge cases', () => {
-
-  it('supports multiple declarations', async () => {
-    // process.env.BASIC is equal to `basic` because the second .env file overrides it.
+  it('supports multiple declarations, including optional ones', async () => {
     const code = `
       const assert = require('assert');
       assert.strictEqual(process.env.BASIC, 'basic');
       assert.strictEqual(process.env.NODE_NO_WARNINGS, '1');
     `.trim();
     const child = await common.spawnPromisified(
       process.execPath,
-      [ `--env-file=${nodeOptionsEnvFilePath}`, `--env-file=${validEnvFilePath}`, '--eval', code ],
+      [ `--env-file=${nodeOptionsEnvFilePath}`, `--env-file-if-exists=${validEnvFilePath}`, '--eval', code ],
       { cwd: __dirname },
     );
     assert.strictEqual(child.stderr, '');
```

Comment:
`i` will take the following values:

1. `0b00`
2. `0b01`
3. `0b10`
4. `0b11`

`i & 0b01` is truthy when `i` has value `0bx1` (i.e. `0b01` and `0b11`), falsy otherwise.
`i & 0b10` is truthy when `i` has value `0b1x` (i.e. `0b10` and `0b11`), falsy otherwise.

`&` is the [bitwiseAND](https://tc39.es/ecma262/#sec-numeric-types-number-bitwiseAND) operator. It's not really for performance reason, but rather as it was the most straight forward way I could think to list all combinations.

Reply:
Yeah just as I thought, but glad I double checked 👍🏼 (haven't used bitwise operations in a ~bit~ while).

It is fairly unconventional... However, I think a simple comment above it should clear any doubts about what it does, so I'm ok with it.

Reply:
To clarify, this is just a suggestion, if you're more comfortable writing it without bitwise operators, that's certainly fine by me as long as we test all four possibilities.

Reply:
It's all good. Your solution works just fine so we'll roll with it. Pushing changes!